dev/NavigationView/NavigationView.cpp

```diff
@@ -1905,6 +1905,11 @@ void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const
         ChangeSelectStatusForItem(nextItem, true /*selected*/);
         RaiseSelectionChangedEvent(nextItem, isSettingsItem, recommendedDirection);
         AnimateSelectionChanged(nextItem);
+
+        if (auto const nvi = NavigationViewItemOrSettingsContentFromData(nextItem))
+        {
+            ClosePaneIfNeccessaryAfterItemIsClicked(nvi);
```

Comment:
I think it makes more sense to place the call to this at the end of `OnSettingsInvoked`. This way we mirror what we do in `OnNavigationViewItemInvoked` and don't have to have this if condition.

Reply:
Nvm, on second glance I see why this was placed here.

Reply:
`OnSettingsInvoked` is not raised when NavigationView items are selected programmatically. Moving the code in this method would solve the issue when selecting *Settings* but not for all the other cases (i.e. the newly introduced interaction test would fail). The issue which this PR aims to fix uses the `NavigationView.SelectedItem` property to select an item programmatically and as such moving this code into the suggested `OnSettingsInvoked` wold not fix it.